zeth-contracts/contracts/BaseMerkleTree.sol

```diff
@@ -7,76 +7,76 @@ pragma solidity ^0.5.0;
 // Adapted from: https://github.com/zcash-hackworks/babyzoe
 
 contract BaseMerkleTree {
-  // Index of the current node: Index to insert the next incoming commitment
-  uint currentNodeIndex;
+    // Depth of the merkle tree (should be set with the same depth set in the
+    // cpp prover)
+    uint constant depth = 4;
```

Comment:
Just a quick comment on the types. It may be nice - in a second time - to refine the types used in the contracts. For example, here a `uint` for the depth looks quite overkill. In practice the depth would ideally be something like 64, which fits into a `uint8`. The corresponding number of leaves would be 2^64 and so the `currentNodeIndex` could become a `uint64`, and so on and so forth.
Doing so would enable to minimize the number of slots (256-bit words) used for storage (by "packing" multiple variables into the same word) and save gas here and there.

Reply:
Ah yes, there are definitely some things that can be packed.
I would hope that constants don't actually get put into storage, but I'd need to actually check that.

Definitely places where we can pack things, though.